app/lib/shared/configuration.dart

```diff
@@ -255,10 +254,12 @@ class Configuration {
   factory Configuration.fromEnv(EnvConfig env) {
     if (env.gcloudProject == 'dartlang-pub') {
       return Configuration._prod();
-    } else if (env.gcloudProject == 'dartlang-pub-dev') {
-      return Configuration._dev();
+    } else if (env.gcloudProject.startsWith('dartlang-pub-dev')) {
```

Comment:
I think it might be better to have a separate env var to indicate whether we want a development or production profile..

@isoos any thoughts here?

Reply:
I think this is not a pattern that we'd like to establish. If somebody wants to run pub on appengine, they should use their own namespace and naming conventions (for the project, for the buckets, for everything basically).

It is more pronounced with the lack of popularity file, and I could imagine it may happen with more stuff in the future.

I'd rather have a solid no-dependency local development environment for contributors of pub.dev, and a separate pluggable-backend pub server that people can self-host.

Reply:
As a consequence, I think I'll just split the current `development.md` into two, and frame the Appengine setup part as supporting documentation for pub-dev administrators only. Maybe we should also rename these configuration options too. wdyt?

Reply:
> I think this is not a pattern that we'd like to establish. If somebody wants to run pub on appengine, they should use their own namespace and naming conventions (for the project, for the buckets, for everything basically).

currently, the configuration seems to be hardcoded.
Developer need access to `dartlang-pub-dev` and it's subresources (buckets/datastore/...) and this seems impossible for not google/dart member.
To spin up `pub.dev` on my own I had to replace project name refs and debug to find out datastore is selected base on `env.gcloudProject`

Personally, I'd prefer to have logic like:
```
if env.gcloudProject == 'dartlang-pub':
    return config._prod()
else:
    return config._dev(env.gcloudProjec)
```

Reply:
@n0npax: what do you think about using a separate config file? If you want to use your own custom AppEngine project, most of the values in the configuration needs to be updated, and it would be handled better via a config file (e.g. a `.yaml` that is parsed into a `Configuration` object via `json_serializable`). If you would like to do that, I can help you with the change.

At the same time, I'll refactor the documentation around local development a bit. I'm using the fake server locally for a time now, and I think it is ideal for small contributions. I'd like to emphasise that a bit more than currently is. I'd also do a separate documentation on how to run the project on your custom AppEngine project, but that seems to be a bit lower priority for me (compared to the immediately available fake server). If you want to do that setup, and willing to identify the blockers, I'm happy to get you through and update the code and/or the docs.

Reply:
The idea of `yaml` based config files sounds good. 
yes, I'll be happy to do refactor this PR to use it.